dinstaller-lib/src/settings.rs

```diff
@@ -0,0 +1,122 @@
+use crate::users::{UsersClient, FirstUser};
+use crate::storage::StorageClient;
+use std::{str::FromStr, error::Error, default::Default};
+use crate::attributes::{Attributes, AttributeValue};
+
+#[derive(Debug, Default)]
+pub struct Settings {
+    pub user: UserSettings,
+}
+
+#[derive(Debug, Default)]
+pub struct UserSettings {
+    pub full_name: String,
+    pub user_name: String,
+    pub password: String,
+    pub autologin: bool,
+}
+
+#[derive(Debug)]
+pub struct StorageSettings {
+    lvm: bool,
+    encryption_password: String
+}
+
+impl Attributes for Settings {
+    fn set_attribute(&mut self, attr: &str, value: AttributeValue) -> Result<(), &'static str> {
+        if let Some((ns, id)) = attr.split_once(".") {
+            match ns {
+                "users" => {
+                    self.user.set_attribute(id, value)?
+                },
+                _ => return Err("unknown attribute")
+            }
+        }
+        Ok(())
+    }
+}
+
+impl Attributes for UserSettings {
+    fn set_attribute(&mut self, attr: &str, value: AttributeValue) -> Result<(), &'static str> {
+        match attr {
+            "full_name" => self.full_name = value.try_into()?,
```

Comment:
definitively see possibility for macro here.

Reply:
Yes, I agree.